### PR TITLE
allow protocols to contain '-'

### PR DESCRIPTION
**Pull request by julienschmidt**
The lexer currently does not allow protocol names to contain '-'.
However, `tcp-pkt` and `tcp-stream` are valid protocol names in Suricata: https://github.com/OISF/suricata/blob/master/doc/userguide/rules/differences-from-snort.rst#dont-cross-the-streams

### CODE REVIEW

File: lex.go [hunk 1]
```diff
@@ -229,15 +237,16 @@ func lexComment(l *lexer) stateFn {
...
 // lexAction consumes a rule action.
 func lexAction(l *lexer) stateFn {
-	r := l.next()
-	switch {
-	case r == ' ':
-		l.emit(itemAction, true)
-		return lexProtocol
-	case unicode.IsLetter(r):
-		return lexAction
+	for {
```
Comment (julienschmidt) [inline lex.go]: This is unrelated refactoring and should have been a separate commit.

Comment (duanehoward) [reply lex.go]: Do you want to start a new PR without this?

Comment (julienschmidt) [reply lex.go]: It should be fine to leave it in here unless you request some changes.
It's basically the same change applied to the other functions in an earlier PR. I overlooked this one.
